src/cargo/util/hex.rs

```diff
@@ -21,6 +23,19 @@ pub fn hash_u64<H: Hash>(hashable: H) -> u64 {
     hasher.finish()
 }
 
+pub fn hash_u64_file(mut file: &File) -> u64 {
+    let mut hasher = SipHasher::new_with_keys(0, 0);
+    let mut buf = [0; 64 * 1024];
+    loop {
+        let n = file.read(&mut buf).unwrap();
+        if n == 0 {
+            break;
+        }
+        hasher.write(&buf);
```

Comment:
comparing it to https://github.com/rust-lang/cargo/blob/ab2b2c0b05a5ec8f15b17433a56b3a8b70f4cc2a/src/cargo/util/sha256.rs#L20-L29
I'd have expected `hasher.write(&buf[..n]);`

Reply:
Oof, sorry about that.